weasyl/test/web/test_interaction.py

```diff
@@ -0,0 +1,66 @@
+import pytest
+
+from weasyl import errorcode
+from weasyl.test import db_utils
+
+
+INVALID_PATHS = (
+    '/note',
+    '/note?noteid=1&noteid=2',
+)
+
+
+@pytest.mark.usefixtures('db')
+@pytest.mark.parametrize('path', INVALID_PATHS)
+def test_get_note_guest_invalid(app, path):
+    resp = app.get(path, status=403)
+    assert resp.html.find(id='error_content').p.text.strip() == errorcode.unsigned
+
+
+@pytest.mark.usefixtures('db')
+@pytest.mark.parametrize('path', INVALID_PATHS)
+def test_get_note_invalid(app, path):
+    user = db_utils.create_user()
+    cookie = db_utils.create_session(user)
+    resp = app.get(path, headers={'Cookie': cookie}, status=404)
+    assert resp.html.find(id='error_content').p.text.strip() == errorcode.error_messages['noteRecordMissing']
+
+
+@pytest.mark.usefixtures('db')
+def test_get_note_permissions(app):
+    user1 = db_utils.create_user(username='user1')
+    cookie1 = db_utils.create_session(user1)
+    user2 = db_utils.create_user(username='user2')
+    cookie2 = db_utils.create_session(user2)
+    user3 = db_utils.create_user(username='user3')
+    cookie3 = db_utils.create_session(user3)
+
+    app.post('/notes/compose', {
+        'recipient': 'user2',
+        'title': 'Title 1',
+        'content': 'Content 1',
+    }, headers={'Cookie': cookie1})
+
+    app.post('/notes/compose', {
+        'recipient': 'user3',
+        'title': 'Title 2',
+        'content': 'Content 2',
+    }, headers={'Cookie': cookie2})
+
+    resp = app.get('/note?noteid=1', headers={'Cookie': cookie1})
```

Comment:
Can’t rely on these notes having a particular id. As far as how to find that id… exercising `/notes` might not be a bad thing?